### ISSUE TITLE
Questions about gated query transformation

### ISSUE THREAD

**Issue by fzb408** [WillDreamer/Aurora#4]
Thank you for writing such an excellent paper. Where in the code is the operation of the query gate mentioned in your paper？

**Comment by xinlong-yang** [comment]
Thanks for your interest. The relevant part is in 'BertLayer' class in med.py. Since you get 'layer_output' (crossattention output) and 'residual_text' (scale&shift selfattention output), you can use the following code to perform gated query transformation:
```
g = torch.softmax(torch.sum(torch.matmul(layer_output,residual_text.permute(0, 2, 1)),dim=-1),dim=1).unsqueeze_(2)
layer_output = layer_output * g + (1 - g) * residual_text
```

**Comment by fzb408** [comment]
Thank you very much for your help！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "WillDreamer/Aurora"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年11月17日(星期五) 晚上8:01
***@***.***&gt;;
***@***.***&gt;;"State ***@***.***&gt;;
主题:&nbsp;Re: [WillDreamer/Aurora] Questions about gated query transformation (Issue #4)





 
Thanks for your interest. The relevant part is in 'BertLayer' class in med.py. Since you get 'layer_output' (crossattention output) and 'residual_text' (scale&amp;shift selfattention output), you can use the following code to perform gated query transformation:
 g = torch.softmax(torch.sum(torch.matmul(layer_output,residual_text.permute(0, 2, 1)),dim=-1),dim=1).unsqueeze_(2) layer_output = layer_output * g + (1 - g) * residual_text

**Comment by Eureka-JTX** [comment]
> Thanks for your interest. The relevant part is in 'BertLayer' class in med.py. Since you get 'layer_output' (crossattention output) and 'residual_text' (scale&shift selfattention output), you can use the following code to perform gated query transformation:
> 
> ```
> g = torch.softmax(torch.sum(torch.matmul(layer_output,residual_text.permute(0, 2, 1)),dim=-1),dim=1).unsqueeze_(2)
> layer_output = layer_output * g + (1 - g) * residual_text
> ```

Hi~ Thanks for the excellent work. 
So now the codes about **qurey gated transformation** and **Informative Context Enhancement**  are not in the repo?  Should I add my own code implementation?

**Comment by Eureka-JTX** [comment]
> Thank you very much for your help！
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "WillDreamer/Aurora" ***@***.***&gt;; 发送时间:&nbsp;2023年11月17日(星期五) 晚上8:01 ***@***.***&gt;; ***@***.***&gt;;"State ***@***.***&gt;; 主题:&nbsp;Re: [WillDreamer/Aurora] Questions about gated query transformation (Issue #4) Thanks for your interest. The relevant part is in 'BertLayer' class in med.py. Since you get 'layer_output' (crossattention output) and 'residual_text' (scale&amp;shift selfattention output), you can use the following code to perform gated query transformation: g = torch.softmax(torch.sum(torch.matmul(layer_output,residual_text.permute(0, 2, 1)),dim=-1),dim=1).unsqueeze_(2) layer_output = layer_output * g + (1 - g) * residual_text — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you modified the open/close state.Message ID: ***@***.***&gt;

Hi, have you solved it?

**Comment by Eureka-JTX** [comment]
@fzb408  Hi，have you reproduced it successfully？

**Comment by Arsiuuu** [comment]
@fzb408 Hello~ have you reproduced successfully?

**Comment by fzb408** [comment]
> 你好~你复制成功了吗？

您好，我没有复制成功

**Comment by Arsiuuu** [comment]
> > 你好~你复制成功了吗？
> 
> 您好，我没有复制成功

没有复现出结果吗，是图像还是视频

**Comment by fzb408** [comment]
> 新一期

我想要把门限查询应用到另一个代码中没有成功，报出了loss.backward的错误
